palladian-commons/src/test/java/ws/palladian/helper/StopWatchTest.java

```diff
@@ -0,0 +1,150 @@
+    Assert.assertEquals(0L, stopWatch.getCountDown());
+
+    stopWatch.setCountDown(5_432L);
+    Assert.assertEquals(5_432L, stopWatch.getCountDown());
+  }
+
+  @PrepareForTest({System.class, StopWatch.class})
+  @Test
+  public void testGetElapsedTime() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L).thenReturn(0L).thenReturn(0L).thenReturn(4_758L);
+
+    stopWatch.start();
+    stopWatch.stop();
+
+    Assert.assertEquals(0L, stopWatch.getElapsedTime(false));
+    Assert.assertEquals(0L, stopWatch.getElapsedTime(true));
+
+    stopWatch.start();
+    stopWatch.stop();
+
+    Assert.assertEquals(4_758L, stopWatch.getElapsedTime(false));
+    Assert.assertEquals(4L, stopWatch.getElapsedTime(true));
+  }
+
+  @PrepareForTest({DateHelper.class, StopWatch.class, System.class})
+  @Test
+  public void testGetElapsedTimeStringAndIncrement() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    Assert.assertEquals("0ms (+0ms)", stopWatch.getElapsedTimeStringAndIncrement());
+  }
+
+  @PrepareForTest({StopWatch.class, System.class})
+  @Test
+  public void testGetTotalElapsedTimeString() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    stopWatch.stop();
+
+    Assert.assertEquals("0ms", stopWatch.toString());
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(false));
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(true));
+
+    stopWatch.setOutputDetail(null);
+
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString());
+  }
+
+  @PrepareForTest({DateHelper.class, StopWatch.class, System.class})
+  @Test
+  public void testGetTotalElapsedTimeString2() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(0L);
+
+    Assert.assertEquals("0ms", stopWatch.toString());
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(false));
+    Assert.assertEquals("0ms", stopWatch.getTotalElapsedTimeString(true));
+  }
+
+  @Test
+  public void testTimeIsUpOutputFalse() {
+
+    final StopWatch stopWatch = new StopWatch();
+
+    Assert.assertEquals(false, stopWatch.timeIsUp());
+  }
+
+  @PrepareForTest({StopWatch.class, System.class})
+  @Test
+  public void testTimeIsUpOutputFalse2() {
+
+    PowerMockito.mockStatic(System.class);
+
+    final StopWatch stopWatch = new StopWatch();
+    stopWatch.setCountDown(-2L);
+    PowerMockito.when(System.currentTimeMillis()).thenReturn(-765_937L);
+
+    Assert.assertEquals(false, stopWatch.timeIsUp());
+  }
+
+  @Test
+  public void testTimeIsUpOutputTrue() {
```

Comment:
I think this test could be combined with testTimeIsUpOutputFalse()?
Perhaps it can be combined with testTimeIsUpOutputFalse2()? (assuming all 3 are needed)